FB0T-134

KamelaJda opened the pull request.

KamelaJda committed:
FB0T-134 Pierwsze testy (jeszcze 2137 commitow do pelnego wydania)

KamelaJda committed:
FB0T-134 Tlumaczenia? Co to?

KamelaJda committed:
Drobna poprawka

KamelaJda committed:
Glupi null

KamelaJda committed:
Kolejne zmiany

KamelaJda committed:
Kolejne zmiany

KamelaJda committed:
Kolejne zmiany

KamelaJda committed:
Kolejne zmiany

KamelaJda committed:
Teraz powinno byc ok

KamelaJda committed:
Teraz powinno byc ok

KamelaJda committed:
Teraz powinno byc ok

KamelaJda committed:
Teraz powinno byc ok

KamelaJda committed:
Czekaj co

KamelaJda committed:
Czekaj co

KamelaJda committed:
Juz prawie

KamelaJda committed:
Juz prawie

KamelaJda committed:
Juz prawie

KamelaJda committed:
Juz prawie

KamelaJda committed:
Juz prawie

KamelaJda committed:
Juz prawie

KamelaJda committed:
Juz prawie

KamelaJda committed:
Ok, skonczone

fratik commented on commands/src/main/java/pl/fratik/commands/system/PrefixroleCommand.java:
```diff
@@ -0,0 +1,155 @@
+        aliases = new String[] {"roleprefix"};
+        uzycieDelim = " ";
+        LinkedHashMap<String, String> hmap = new LinkedHashMap<>();
+        hmap.put("set|remove|list", "string");
```
mogłeś to zrobić na subkomendach XD

KamelaJda replied:
nie umiem w subkomendy XD

fratik replied:
Przykład masz w FB0T-102 xD

KamelaJda committed:
Merge branch 'v3' into FB0T-134

fratik commented on commands/src/main/java/pl/fratik/commands/MemberListener.java:
```diff
@@ -115,6 +112,46 @@ public void onMessage(MessageReceivedEvent m) {
+            AtomicReference<String> nick = new AtomicReference<>(mem.getNickname());
+            if (nick.get() == null) nick.set(mem.getUser().getName());
+            else {
+                gc.getRolePrefix().values().forEach(p -> {
```
zastąpiłbym `for (... : gc.getRolePrefix().values())`, nie trzeba by było kombinować z AtomicReference

fratik commented on commands/src/main/java/pl/fratik/commands/MemberListener.java:
```diff
@@ -115,6 +112,46 @@ public void onMessage(MessageReceivedEvent m) {
+                mem.getGuild().modifyNickname(mem, prefix + " " + nick.toString()).queue();
+                return;
+            } catch (Exception e) {
+                /* brak permów */
```
imo jeszcze przydałoby się puścić info np na pełne logi, by potem nie było "eJ TO NEI DZIAŁA :CCCCC"

fratik commented on commands/src/main/java/pl/fratik/commands/MemberListener.java:
```diff
@@ -115,6 +112,46 @@ public void onMessage(MessageReceivedEvent m) {
+            else nick.set(nick.toString());
+            if ((prefix + nick).length() > 31) {
+                int dlugosc = (prefix + nick).length() - 31;
+                nick.set(nick.toString().substring(dlugosc));
```
to jest ucięcie nicku? jak tak to nie lepiej dać trzy kropki?

KamelaJda replied:
yyy, myslalem tak: `jezeli ktos bedzie mial za dlugi prefix + nick to niech mu przycina nick`

fratik replied:
To nie lepiej jak jest w fdevie: „bardzo długi ni...”?

KamelaJda replied:
imo nie, lepiej mieć końcówkę nicku niż `...`

fratik committed:
FB0T-134 poprawki

fratik committed:
Merge branch 'v3' into FB0T-134

fratik committed:
FB0T-134 poprawki

fratik committed:
FB0T-134 zabieranie prefixu po utracie roli (?)

fratik committed:
jFB0T-134 tak

fratik committed:
poprawki

fratik committed:
FB0T-134 substr na odwrót

fratik committed:
FB0T-134  poprawki

poprawione teksty, ZWS zamiast pustego stringa w fieldach,

lepsza kalkulacja dlugosci fieldow, automatyczna aktualizacja nicków

fratik committed:
FB0T-134 jeżeli always true to always true, Kamil

fratik committed:
FB0T-134 wsparcie dla więcej niż jednego słowa

powodzenia na 8 znakach btw

no chyba że ktoś chce napisać jebać pis

jak ktoś tego commita przeczyta w jakimś artykule to pozdrawiam serdecznie <3

fratik committed:
prefixrole -> prefixroli

fratik committed:
FB0T-134 najpierw save, później update
